### PR TITLE
Homework 1 + tests for homework1

### PR DESCRIPTION
**Pull request by featherko**

### CODE REVIEW

File: homework/hw1/task3_minmax.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        lines = fi.readlines()
+        min_value = int(lines[0])
+        max_value = int(lines[0])
+        for line in lines:
+            line = int(line)
+            if line > max_value:
+                max_value = line
+            elif line < min_value:
+                min_value = line
+        return min_value, max_value
```
Comment (Aleksei-Samarin) [inline homework/hw1/task3_minmax.py]: There is no need in keeping old solutions here and below, you can remove them to beautify your modules

File: homework/hw1/task2_fib.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+def check_fibonacci(data: Sequence[int]) -> bool:
+    """Check fibonacci.
+
+    Checks if given sequence is fibonacci sequence
+
+    :param data:  given sequence
+    :return:  Returns true if given sequence is fibonacci
+    """
+    if len(data) <= 2:
+        return data == [0, 1]
```
Comment (Aleksei-Samarin) [inline homework/hw1/task2_fib.py:21]: It's OK, but what if we got [0]?

File: homework/hw1/task4_sum41.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    Checks how many tuples i, k, k, l there are such chat A[i], B[j], C[k], D[l] is zero
+
+    :param a: First list (A)
+    :param b: Second List (B)
+    :param c: Third List (C)
+    :param d: Forth List (D)
+    :return: Number of such tuples
+    """
+    return sum(1 for x in itertools.product(a, b, c, d) if sum(x) == 0)
```
Comment (Aleksei-Samarin) [inline homework/hw1/task4_sum41.py:23]: Great realisation! Beauty of Python in example

File: test/hw1/test_task5_maxsub.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from typing import List, NoReturn, Optional
+
+import pytest
+
+from homework.hw1.task5_maxsub import find_maximal_subarray_sum
+
+
+@pytest.mark.parametrize(
+    ("array", "k", "expected_result"),
+    [([1, 2, 3, 4, 5], 1, 5), ([-1, -2, -3], 1, -1), ([-1, -2, 3], 1, 3)],
```
Comment (Aleksei-Samarin) [inline test/hw1/test_task5_maxsub.py:12]: For more readability these tuples can be placed on different lines

**Review by Aleksei-Samarin — CHANGES_REQUESTED** [review]

File: homework/hw1/task2_fib.py (hunk 2 above)
Comment (featherko) [reply homework/hw1/task2_fib.py:21]: Should we consider [0] as sequence? It's just a number, thought it's part of sequence.  My idea was to check if given sequence is fib sequence, and therefore if it's just a number answer should be false.

File: test/hw1/test_task5_maxsub.py (hunk 4 above)
Comment (featherko) [reply test/hw1/test_task5_maxsub.py:12]: black . formating turns them like this, since they don't exceed max length of line

**Pull request merged**